### ISSUE TITLE
pinner infra-structure

### ISSUE THREAD

**Issue by pgte** [ipfs-shipyard/peer-star-app#193]
...continued from #189.

Alternatives on the table (from [@parkan 's comment](https://github.com/ipfs-shipyard/peer-star-app/issues/189#issuecomment-444620243))

> ways I can think of solving this in a more decentralized-first manner:

> * pinners present their identity via HD certificate (for example official PL PeerPad pinners have certificates derived from the https://peerpad.net SSL cert)
> * pinner is paid in filecoin, status shows $FC balance for the collab
> * pinner must be explicitly invited by participant holding keys

> options 1 and 2 are not entirely practical today, but I wonder if pinner access can be restricted in some simpler way for now

**Comment by aschmahmann** [comment]
TLDR: Users should trust a set of pinners that is configurable by them.

I think the general question being asked here is how do we translate, or generalize, the semantics of the server/cloud world to the distributed world. Some of these semantics are going to be hard to translate.

For persistence indicators we are effectively trying to translate "[Insert Cloud Provider] says they have my data" into "The people that I care about have my data". Who those people are though is unclear, some options, related to the suggestions of @parkan and @jimpick, which can be combined are:

1. Some set of people that I want to have my data have confirmed having it (e.g. 6/10 users I want to share the data with have it)
2. A centralized paid service confirms having my data (e.g. my personal pinner or trusted infrastructure)
3. A decentralized service confirms having my data (e.g. the data is in a blockchain)

In the short-medium term I think options 1 and 2 are the easiest to get off the ground, however given that this discussion assumes the existence of pinners it seems the best option is 2. For example, in open-collaborations like PeerPad there is no strong sense of "users I want to have the data", since you are theoretically sharing with the whole network. In this case, requiring users to trust their preferred pinning services to be globally available whether they are A) paid for, or B) provided for free by some organization, seems fine.

I think that it's reasonable to assume that as we transition from centralized web services to distributed services there will be plenty of required intermediaries (like pinning services) along the way. For example, IPFS pinners may reasonably be replaced by Filecoin, but I am unaware of any work towards creating a decentralized network for replacing IPNS pinners (i.e. republishers). To that end, supporting the use case of trusted "pinners" that can always be swapped out for other pinners (e.g. switching from an AWS-based pinner to an Azure-based pinner should be trivial) seems quite important.
